hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -439,4 +439,87 @@ static function &importableFields($contactType = 'Individual',
       self::$_importableFields = $fields;
     return self::$_importableFields;//$fields;
   }
+
+  /**
+   * Returns a list of active contracts.
+   *
+   * If no startDate is given, the current date will be used. If no endDate is
+   * given, the startDate will be used for it.
+   *
+   * A contract is active if:
+   * - The effective date of current revision is less or equal the startDate OR
+   *   it is starts someday between the start and end dates
+   * - The contract start date and end dates overlaps with the start and end
+   *   dates passed to the period OR
+   *   it doesn't have an end date and starts before the given start date OR
+   *   it doesn't have an end date and starts between the given start and end
+   *   dates
+   * - The contract is not deleted
+   *
+   * @param null $startDate
+   * @param null $endDate
+   *
+   * @return array
+   */
+  public static function getActiveContracts($startDate = null, $endDate = null)
+  {
+    if($startDate) {
+      $startDate = CRM_Utils_Date::processDate($startDate, null, false, 'Y-m-d');
+    } else {
+      $startDate = date('Y-m-d');
+    }
+
+    if($endDate) {
+      $endDate = CRM_Utils_Date::processDate($endDate, null, false, 'Y-m-d');
+    } else {
+      $endDate = $startDate;
+    }
+
+    $query = "
+      SELECT c.*
+      FROM civicrm_hrjobcontract c
+        INNER JOIN civicrm_hrjobcontract_revision r
+          ON r.id = (SELECT id
+                     FROM civicrm_hrjobcontract_revision r2
+                     WHERE
+                      r2.jobcontract_id = c.id AND
+                      (
+                        r2.effective_date <= '$startDate'
```

Comment:
Please enclose variables in {}.
So for example, $startDate will be {$startDate}